File: src/등수구하기.java
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class 등수구하기 {

    static String solution(int n, int[] k) {
        String answer = "";

        for (int i = 0; i < n; i++) {
            int tmp = 1;
            for (int j = 0; j < n; j++){
                if(k[i] < k[j]) tmp++;
            }
            answer += tmp+" ";
        }

        return answer;
    }

    public static void main(String[] args) throws Exception {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(bufferedReader.readLine());
        String[] str = bufferedReader.readLine().split(" ");
        int[] k = new int[n];

        for (int i = 0; i < n; i++) {
            k[i] = Integer.parseInt(str[i]);
        }

        System.out.println(solution(n, k));
    }
}

File: src/등수구하기.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main() {
    val bufferedReader = BufferedReader(InputStreamReader(System.`in`))
    val n = bufferedReader.readLine().toInt()
    val s = bufferedReader.readLine().split(" ")
    val k = IntArray(n)

    for(i in 0 until n){
        k[i] = s[i].toInt()
    }

    println(solution(n, k))
}

fun solution(n : Int , k : IntArray) : String{
    var answer = ""

    for (i in 0 until n){
        var tmp = 1
        for (j in 0 until n){
            if( k[i] < k[j]) tmp++
        }
        answer += "$tmp "
    }

    return answer
}